ext/standard/array.c

```diff
@@ -2475,11 +2475,14 @@ static void php_compact_var(HashTable *eg_active_symbol_table, zval *return_valu
 			Z_PROTECT_RECURSION_P(entry);
 		}
 		ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(entry), value_ptr) {
-			php_compact_var(eg_active_symbol_table, return_value, value_ptr);
+			php_compact_var(eg_active_symbol_table, return_value, value_ptr, pos);
 		} ZEND_HASH_FOREACH_END();
 	    if (Z_REFCOUNTED_P(entry)) {
 			Z_UNPROTECT_RECURSION_P(entry);
 		}
+	} else {
+		zend_throw_error(zend_ce_type_error, "Parameter %d in compact() must be string or array of strings", pos);
```

Comment:
```suggestion
		zend_throw_error(zend_ce_type_error, "Parameter %"PRIu32" in compact() must be string or array of strings", pos);
```
for the correct format. I'm actually surprised CI didn't complain.

Actually even better would be:
```suggestion
		zend_argument_type_error(pos, "must be string or array of strings, , %s given", zend_zval_type_name(entry));
```
For the usual standard formatting.